### ISSUE TITLE
NPM 7 fails with an empty script.

### ISSUE THREAD

**Issue by mshima** [npm/cli#2171]
### Current Behavior:
NPM 7 fails with an empty script.

### Expected Behavior:
Not to fail.

### Steps To Reproduce:
package.json:
```
{
  "scripts": {
    "empty": ""
  }
}
```

Command
```
npm run empty
```

Result:
```
npm ERR! missing script: empty
npm ERR!
npm ERR! Did you mean this?
npm ERR!     empty
```

### Environment:

- OS: MacOS 10.15.7
- Node: 14.15.0
- npm: 7.0.10

**Comment by ljharb** [comment]
What does npm 6 do here?

**Comment by mshima** [comment]
> What does npm 6 do here?

Doesn’t fails.

**Comment by mshima** [comment]
Workaround is to add a space:
```
{
  "scripts": {
    "empty": " "
  }
}
```

**Comment by nlf** [comment]
this is technically a bug, but i'm curious about the use case for having empty scripts and expecting to be able to run them.. is the idea to make sure that the command always works?

if so: `npm run --if-present empty` will probably do what you want

**Comment by ljharb** [comment]
Seems like any script that trims to the empty string should be treated as an error case, distinct from missing scripts - I’m curious about the use case for an empty script.

**Comment by nlf** [comment]
my guess was a build system of some sort that runs `npm run empty` for every project it builds, which is why i suggested the `--if-present` flag since that would run the script if it exists and do nothing at all if it doesn't

**Comment by mshima** [comment]
> Seems like any script that trims to the empty string should be treated as an error case, distinct from missing scripts - I’m curious about the use case for an empty script.

Not against it but the error message should be fixed.
At jhipster we generate a bunch of scripts, with pre scripts.
In some cases we generate some empty scripts, because that script does not apply.
To keep scripts 'api' stable with pre scripts, the both are generated blank. `--if-present` is a little more error prone for our use case. If the generator failed to generate the script, CI will still pass.

**Comment by nlf** [comment]
fair enough :+1:

pull request to fix it is up and will hopefully land in the next release

**Comment by mshima** [comment]
Thanks.

**Comment by nlf** [comment]
this was resolved in 7.0.12
